src/EditorFeatures/CSharpTest/UseNullPropagation/UseNullPropagationTests.cs

```diff
@@ -478,5 +478,125 @@ void Main(S? s)
 }
 ");
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseNullPropagation)]
+        public async Task TestWithNullableTypeAndIsNull()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    public int? f;
+    void M(C c)
+    {
+        int? x = [||]c is null ? null : c.f;
+    }
+}",
+@"
+class C
+{
+    public int? f;
+    void M(C c)
+    {
+        int? x = c?.f;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseNullPropagation)]
+        public async Task TestWithNullableTypeAndReferenceEquals()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    public int? f;
+    void M(C c)
+    {
+        int? x = [||]ReferenceEquals(c, null) ? null : c.f;
```

Comment:
Cool changes :)

Consider also supporting:

```!ReferenceEquals(c, null) ? c.f : null```

Reply:
That’s a good idea. I’ll look into that later tonight.

Reply:
:bulb: Also have an opportunity here for `!(c is null)`

Reply:
It was quite simple to implement this so that logical negation would be recognized on any condition, including `!(c == null)` and `!(c != null)`. If to only support it on the `ReferenceEquals` and pattern syntax I can change it.